main.go

```diff
@@ -326,18 +325,14 @@ func runWithOptions(opts Options, out output, logger *zap.Logger) {
 	serializer = withTransportSerializer(transport.Protocol(), serializer, opts.ROpts)
 
 	// req is the transport.Request that will be used to make a call.
-	req, err := serializer.Request(reqInput)
+	rawReq, err := serializer.Request(reqInput)
```

Comment:
nit: any reason to call this `rawReq` instead of just `req`. They're both the same type.

Reply:
i'm not a big fan of reassigning/mutating data in general, which is why i'm generally pretty explicitly about it throughout this PR

Reply:
Yeah, except that's unavoidable in this situation. Mutation is part of the
lifecycle of the object here. The two references are the same. For all cases,
`req == rawReq` so there's no reason for them to be two separate variables.